piptools/utils.py

```diff
@@ -121,6 +122,11 @@ def format_requirement(
     elif is_url_requirement(ireq):
         line = _build_direct_reference_best_efforts(ireq)
     else:
+        # Canonicalize the requirement name
+        # https://packaging.pypa.io/en/latest/utils.html#packaging.utils.canonicalize_name
+        req = copy.copy(ireq.req)
+        req.name = canonicalize_name(req.name)
+        line = str(req)
```

Comment:
Next line can be removed

Reply:
yeah, and a `copy` import. i trust that @di will also notice and fix this.  (thanks for fixing this btw!)

Reply:
That's what I get for being lazy!

Reply:
no worries, i guess me (or another internet rando) blindly pushing to ‘your’ branch while you're working on it would've caused more mayhem 🙃